### DISCUSSION TITLE
Compiler version must be fixed

### DISCUSSION THREAD

**Discussion by lalondelalonde** [Cyfrin/foundry-full-course-cu#75] [Q&A]
Hello guys, awesome course!

- Currently at lesson 6. Foundry ran properly.
- Forge, Anvil, Cast and Chisel are installed.
- The issue occur when I copy paste SimpleStorage.sol in src:

```
4:1: SyntaxError: Source file requires different compiler version (current compiler is 0.4.17+commit.bdeb9e52.Emscripten.clang - note that nightly builds are considered to be strictly less than the released version\npragma solidity 0.8.19;\n^---------------------^\n",
	"startLineNumber": 4,
	"startColumn": 2,
	"endLineNumber": 4,
	"endColumn": 2
}]
```
I also have 
```
Compiler version must be fixed.
```
It's been a couple of hours that I'm stuck on this and can't seems to find what I must do.

Tried to reinstalled solc but everytime I do so and look for the version I receive: 
```
Command not found.
```
Not sure but since Foundry seems to have an integrated compiler (I'm beginning guys) I think that might be the reason why. I'm also running the code on a MacBook Pro High Sierra which I think might be the issue.

Any suggestions?

**Comment by PatrickAlphaC** [comment]
Thanks for making this here!

Could you send us the exact commands you ran?

That error typically means you need to change your `pragma solidity` version at the top of your contract.

Can you do the following:

1. What does your `SimpleStorage.sol` contract look like? Paste it here with surrounding backticks (```)
2. What does your directory setup look like? Run this command from the project root directory and paste the output here with surrounding backticks (```)
_The project root directory is the "main" folder that has all your files. If you're following the video, this is your `foundry-simple-storage-f23` folder_
```bash
cd foundry-simple-storage-f23
tree .
```

**Comment by lalondelalonde** [comment]
Sure!

The exact command I ran is : forge create SimpleStorage --interactive

I then ran into this first error message :
```
[⠘] Compiling...
[⠊] Installing solc version 0.8.19
[⠰] Successfully installed solc 0.8.19
Error: 
Solc Error: dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/maximelalonde/.svm/0.8.19/solc-0.8.19 (which was built for Mac OS X 11.6)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/maximelalonde/.svm/0.8.19/solc-0.8.19 (which was built for Mac OS X 11.6)
  Expected in: /usr/lib/libSystem.B.dylib
  ```
  I then went to read the flagged problems in my problem window in the terminal of VS Code which I've listed above.
  
  Directory if not mistaken (my folder is on my desktop, but since I can't save/compile cd keep on telling me no such file or directory)
  ```
 MacintoshHD/Users/maximelalonde/Desktop/foundry-f23/foundry-simple-storage-f23
 ```
 
 Thanks!

**Comment by lalondelalonde** [comment]
Hello guys,

I found the issue. The version of my macOS wasn't compatible with the Solidity Compiler.

Installed a newest version and it went through. 

Back to learning , thanks!
